File: go/p26.go
package main

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	currentIndex := 0
	for i := 1; i < len(nums); i++ {
		if nums[i] != nums[currentIndex] {
			nums[currentIndex + 1] = nums[i]
			currentIndex++
		}
	}
	return currentIndex + 1
}

func main() {
	println(removeDuplicates([]int{10, 9, 2, 1, 1}))
}

File: python/p26.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        currentIndex = 0
        for i in range(1, len(nums)):
            if nums[i] != nums[currentIndex]:
                nums[currentIndex + 1] = nums[i]
                currentIndex += 1
        return currentIndex + 1
